writer/writer.ts

```diff
@@ -0,0 +1,72 @@
+import { Chroma } from "@langchain/community/vectorstores/chroma";
+import { getCourseDataForChroma, type ChromaDocumentData } from "../fetcher/course-document-fetcher";
+import type { Document } from "@langchain/core/documents";
+import { OpenAIEmbeddings } from "@langchain/openai";
+import fs from 'fs';
+import path from 'path';
+import dotenv from 'dotenv';
+
+// Load environment variables from .env file
+dotenv.config({ path: path.resolve(__dirname, '../.env') });
+
+const embeddings = new OpenAIEmbeddings({
+    model: "text-embedding-3-small",
+    apiKey: process.env.OPENAI_API_KEY || ""
+});
+
+const vectorStore = new Chroma(embeddings, {
+    collectionName: "uw-courses-test",
+    url: "http://localhost:8080", // Optional, will default to this value
+    collectionMetadata: {
+      "hnsw:space": "cosine",
+    }, // Optional, can be used to specify the distance method of the embedding space https://docs.trychroma.com/usage-guide#changing-the-distance-function
+  });
+  
+const dataFilePath = path.join(__dirname, '../', 'course-data.json');
+
+async function fetchAndStoreCourseData() {
+    console.log("Fetching course data...");
+    const courseData = await getCourseDataForChroma();
+    console.log("Course data fetched:", courseData);
+
+    fs.writeFileSync(dataFilePath, JSON.stringify(courseData, null, 2));
```

Comment:
is there a reason to write to a json file and not just do this in-place?

Reply:
Yes, this makes it so that we don't have to keep calling the waterloo API for the same results. These results will be the same for the next while, so the fetching process is a bit faster.

Reply:
ok